pytest/lib/bridge.py

```diff
@@ -150,51 +277,45 @@ def start(self):
                 self.config_dir,
                 'logs/eth2near-relay/err.log'),
             'a')
-        cli_dir = os.path.join(self.bridge_dir, 'cli')
-        args = ('node index.js start eth2near-relay --daemon false').split()
-        self.pid.value = subprocess.Popen(
-            args, stdout=self.stdout, stderr=self.stderr, cwd=cli_dir).pid
-        # TODO ping and wait until service really starts
-        time.sleep(5)
-        assert self.pid.value != 0
-
-
-class JSAdapter:
-
-    def __init__(self, config):
-        self.config = config
-        self.bridge_dir = self.config['bridge_dir']
-        self.cli_dir = os.path.join(self.bridge_dir, 'cli')
-
-    def call(self, args):
-        if not isinstance(args, list):
-            args = [args]
-        args = ['node', 'index.js', 'TESTING'] + args
-        # TODO check for errors
-        return subprocess.check_output(
-            args, cwd=self.cli_dir).decode('ascii').strip()
+        self.pid.value = self.adapter.js_call_service('eth2near-relay', self.stdout, self.stderr).pid
+        if self.pid.value == 0:
+            logger.error('Cannot start Eth2NearBlockRelay')
+            assert False
 
 
-def assert_deployed(output):
-    assert output.decode('ascii').strip().split(
-        '\n')[-1].strip().split(' ')[0] == 'Deployed'
+def assert_success(message):
+    try:
+        message = message.decode('ascii')
+    except:
+        # message is already str
+        pass
+    if message == '':
+        # empty message is considered valid
+        return
+    message = message.strip().split('\n')[-1].strip().split(' ')
+    if not (message[0] in ('Deployed', 'Done') or message[-1] in ('deployed', 'initialized')):
```

Comment:
The function `assert_success` feels too hacky.

Reply:
You are right. The output of different services is very different. It seems more friendly to have several 
```
assert_success(A) 
assert_success(B)
assert_success(C)
```
instead of
```
assert A.decode('ascii').strip().split('\n')[-1].strip().split(' ')[0] == 'Done'
assert B.decode('ascii').strip().split('\n')[-1].strip().split(' ')[-1] == 'deployed'
assert C == b''
```
So I expect it's better to delegate inner logic to such function and make it a little bit hacky for unification purposes.

Reply:
I'll try to fix in the future if this become problematic.